sasmodels/kerneldll.py

```diff
@@ -265,22 +268,33 @@ def make_dll(source, model_info, dtype=F64):
     # when multiple versions of the application are installed.
     model_file = model_info.id + "_" + generate.tag_source(source)
     dll = dll_path(model_file, dtype)
+    logging.debug("make_dll: dll located %s as %s in %s",
+                  model_info.id, model_file, dll)
 
     if not os.path.exists(dll):
         # Make sure the DLL path exists.
         if not os.path.exists(SAS_DLL_PATH):
             os.makedirs(SAS_DLL_PATH)
-        basename = splitext(os.path.basename(dll))[0] + "_"
-        system_fd, filename = tempfile.mkstemp(suffix=".c", prefix=basename)
         source = generate.convert_type(source, dtype)
-        with os.fdopen(system_fd, "w") as file_handle:
-            file_handle.write(source)
+        if not system:
+            basename = splitext(os.path.basename(dll))[0] + "_"
+            system_fd, filename = tempfile.mkstemp(suffix=".c", prefix=basename)
+            logging.debug("make_dll: writing C for dll: %s", filename)
+            with os.fdopen(system_fd, "w") as file_handle:
+                file_handle.write(source)
+        else:
+            filename = splitext(dll)[0] + ".c"
+            logging.debug("make_dll: writing C for system dll: %s", filename)
+            with open(filename, 'w') as file_handle:
+                file_handle.write(source)
```

Comment:
Is it possible that we differnt sasmodels versions will share the same "compiled" cache? That is, do we need to make sure that  the filename is tagged with version number?

Reply:
It's only precompiled models that have the name change to be predictable but I would expect people who are generating precompiled models (like me!) because they are redistributing sasmodels in precompiled form to deal with this for the user. For instance, the Debian packaging automatically makes sure that there's only ever one version of the precompiled models around and that it matches the version of the python module that is installed.